internal/extension/reconciler_test.go

```diff
@@ -135,3 +247,530 @@ func TestNilGuardedPointer(t *testing.T) {
+//	        │                       │ │              │ │              │ │                       │ │                       │ │                       │      │              │
+//	        │ ┌────────┐ ┌────────┐ │ │ ┌────────┐   │ │  ┌────────┐  │ │ ┌────────┐ ┌────────┐ │ │ ┌────────┐ ┌────────┐ │ │ ┌────────┐ ┌────────┐ │      │ ┌────────┐   │
+//	        │ │ rule-1 │ │ rule-2 │ │ │ │ rule-1 │   │ │  │ rule-1 │  │ │ │ rule-1 │ │ rule-2 │ │ │ │ rule-1 │ │ rule-2 │ │ │ │ rule-1 │ │ rule-2 │ │      │ │ rule-1 │   │
+//	        │ └────┬───┘ └─────┬──┘ │ │ └────┬───┘   │ │  └───┬────┘  │ │ └─┬──────┘ └───┬────┘ │ │ └───┬────┘ └────┬───┘ │ │ └─┬────┬─┘ └────┬───┘ │      │ └────┬───┘   │
+//	        │      │           │    │ │      │       │ │      │       │ │   │            │      │ │     │           │     │ │   │    │        │     │      │      │       │
+//	        └──────┼───────────┼────┘ └──────┼───────┘ └──────┼───────┘ └───┼────────────┼──────┘ └─────┼───────────┼─────┘ └───┼────┼────────┼─────┘      └──────┼───────┘
+//	               │           │             │                │             │            │              │           │           │    │        │                   │
+//	               │           │             └────────────────┤             │            │              └───────────┴───────────┘    │        │                   │
+//	               ▼           ▼                              │             │            │                          ▼                ▼        │                   ▼
+//	┌───────────────────────┐ ┌────────────┐          ┌───────┴─────────────┴───┐  ┌─────┴──────┐             ┌────────────┐        ┌─────────┴──┐          ┌────────────┐
+//	│                       │ │            │          │       ▼             ▼   │  │     ▼      │             │            │        │         ▼  │          │            │
+//	│ ┌────────┐ ┌────────┐ │ │ ┌────────┐ │          │   ┌────────┐ ┌────────┐ │  │ ┌────────┐ │             │ ┌────────┐ │        │ ┌────────┐ │          │ ┌────────┐ │
+//	│ │ port-1 │ │ port-2 │ │ │ │ port-1 │ │          │   │ port-1 │ │ port-2 │ │  │ │ port-1 │ │             │ │ port-1 │ │        │ │ port-1 │ │          │ │ port-1 │ │
+//	│ └────────┘ └────────┘ │ │ └────────┘ │          │   └────────┘ └────────┘ │  │ └────────┘ │             │ └────────┘ │        │ └────────┘ │          │ └────────┘ │
+//	│                       │ │            │          │                         │  │            │             │            │        │            │          │            │
+//	│       service-1       │ │  service-2 │          │         service-3       │  │  service-4 │             │  service-5 │        │  service-6 │          │  service-7 │
+//	└───────────────────────┘ └────────────┘          └─────────────────────────┘  └────────────┘             └────────────┘        └────────────┘          └────────────┘
+func BuildComplexGatewayAPITopology(funcs ...func(*GatewayAPIResources)) GatewayAPIResources {
```

Comment:
This is from `policy-machinery`... 😢 
It just is really tedious to get a dag and I didn't know how to reuse that bit... so `Ctrl+C` & `Ctrl+V` myself outta that